tools/flakiness_checker.py

```diff
@@ -0,0 +1,108 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+""" Checks a given test for flakiness
+Takes the file name and function name of a test, as well as, optionally,
+the number of trials to run and the random seed to use
+"""
+
+import subprocess
+import sys
+import os
+import random
+import argparse
+import re
+import logging
+
+logging.basicConfig(level=logging.INFO)
+
+DEFAULT_NUM_TRIALS = 10000
+DEFAULT_VERBOSITY = 2
+
+def run_test_trials(args):
+    test_path = args.test_path + ":" + args.test_name
+    logging.info("Testing: %s", test_path)
+
+    new_env = os.environ.copy()
+    new_env["MXNET_TEST_COUNT"] = str(args.num_trials)
+    
+    if args.seed is None:
+        logging.info("No test seed provided, using random seed")
+    else:
+        new_env["MXNET_TEST_SEED"] = str(args.seed)
+
+    verbosity = "--verbosity=" + str(args.verbosity)
+
+    code = subprocess.call(["nosetests", verbosity, test_path], 
+                           env = new_env)
+    
+    logging.info("Nosetests terminated with exit code %d", code)
+
+def find_test_path(test_file):
+    """Searches for the test file and returns the path if found
+    As a default, the currend working directory is the top of the search.
+    If a directory was provided as part of the argument, the directory will be
+    joined with cwd unless it was an absolute path, in which case, the
+    absolute path will be used instead. 
+    """
```

Comment:
I don't know if the behaviour described in this doc is actually resembled by the code. I'm a bit concerned about the absolute path because you always join with cwd. Also you should only append the .py if it isn't present yet

Reply:
Path.join will only use the last absolute path provided, so if the user provides an absolute path it will override cwd, I've tested this and it does work this way. Currently, I'm using re.split to process the command line argument with test file and test name, and I'm removing .py from the case when it's present.

Reply:
Awesome! Very nice catch and approach!